### ISSUE TITLE
data_suffix being appened to data_file name when it's None / False.

### ISSUE THREAD

**Issue by amanagr** [coveragepy/coveragepy#989]
When we set `data_suffix` to `False` / `None`, data_file's is appended with a suffix like `.coverage.ubuntu-bionic.2681.658565`. The documentation (https://coverage.readthedocs.io/en/coverage-5.1/api_coverage.html?highlight=data_suffix) says, suffix will only be appended when it's value is `True`, which is not the case here.
Setting `data_suffix` to `""` (empty string) appears to work though.

coverage version = 5.1

**Comment by nedbat** [comment]
From your linked commit, you have `concurrency='multiprocessing'`, which is what forces the suffix.  Don't you need a suffix on the data file? Aren't you recording coverage from a number of processes?

**Comment by amanagr** [comment]
@nedbat I am sorry, but I don't get your point here. I don't want the final data file to have any suffix. I want it named as `.coverage` so that `codecov` can easily find it and upload the coverage results. We are recording coverage from a number of processes.

Am I correct in saying that each process generates a different `.coverage.SUFFIX` file and we can merge them using `.combine` on data? If so, then either the `.combine` or `.save` should have an argument to specify the file name / data_suffix.

**Comment by amanagr** [comment]
Also, can you please tell if setting "data_suffix=""` will pose any problems here? (Or, if there is a better solution for what I want to do there.)

**Comment by nedbat** [comment]
When running with multiple processes, each process needs to write its data to a different file.  The data suffix is what keeps the files separate.  Then when the tests are done, "coverage combine" is used to combine all the files together into a suffix-less file.

It's possible that some of your data is not being collected properly if you are not using suffixes on your data files.  Are you using "coverage combine"?

**Comment by nedbat** [comment]
Sorry, I re-read your comment.  If you are already using .combine, then everything should work properly if you simply don't specify data_suffix at all.  Your "concurrency=multiprocessing" argument will create the correct data files (with a suffix), and then .combine will combine them into a suffix-less file, ready for codecov.

**Comment by amanagr** [comment]
Even with`cov.combine`, I see the final file named ` .coverage.ubuntu-bionic.20843.916684`
where `cov = coverage.Coverage(config_file="tools/coveragerc", concurrency='multiprocessing')`.

My coveragerc:

```
[report]
# Regexes for lines to exclude from consideration
exclude_lines =
    # Re-enable the standard coverage pragma
    nocoverage
    # Don't complain if non-runnable code isn't run:
    if False:
    # Don't require coverage for base class NotImplementedErrors
    raise NotImplementedError()
    # Don't require coverage for test suite AssertionError -- they're usually for clarity
    raise AssertionError
    # Don't require coverage for __str__ statements just used for printing
    def __str__[(]self[)] -> .*:
    # Don't require coverage for errors about unsupported webhook event types
    raise UnexpectedWebhookEventType
    # Don't require coverage for blocks only run when type-checking
    if TYPE_CHECKING:
    # Don't require coverage for abstract methods; they're never called.
    @abstractmethod
    # Don't require coverage for the settings.LOG_API_EVENT_TYPES code paths
    # These are only run in a special testing mode, so will fail normal coverage.
    if settings.LOG_API_EVENT_TYPES:
    # PEP 484 overloading syntax
    ^\s*\.\.\.

[run]
omit =
    */zulip-venv-cache/*
    */migrations/*
    */management/commands/*
    # Parts of the test runner infrastructure
    tools/test-backend
    zerver/lib/test_fixtures.py
    zerver/lib/test_runner.py
    # Has its own independent test suite
    zerver/openapi/python_examples.py
    zerver/lib/parallel.py
    # Debugging tools that don't lend themselves well to unit tests
    zerver/lib/debug.py
    # Part of provisioning/populate_db
    zerver/lib/generate_test_data.py
    # Excluded because its coverage state is flaky.
    zerver/tornado/ioloop_logging.py
    # Zulip's library for use in scripts
    scripts/lib/zulip_tools.py
    # Used only for the legacy Zephyr integration, and unlikely to ever be unit-tested
    zerver/lib/ccache.py
    # Settings.py files are hard to test
    zproject/*settings.py
    # https://github.com/davidhalter/jedi/issues/1122
    blub
```

**Comment by nedbat** [comment]
Can you make a small example that duplicates how you are using the coverage API that shows the problem?

**Comment by amanagr** [comment]
[test-backend](https://github.com/zulip/zulip/blob/master/tools/test-backend) has the complete code.

Code which saves data file with suffix:
```python
cov = coverage.Coverage(config_file="tools/coveragerc", concurrency='multiprocessing')
cov.start()
cov.stop()
cov.combine()
cov.save()
```
You can take `coveragerc` same as my last comment.
Setting `cov = coverage.Coverage(data_suffix="", config_file="tools/coveragerc", concurrency='multiprocessing')` removes the suffix.

**Comment by nedbat** [comment]
We can't change `suffix=None` to suppress the suffix, since None is the default.  But in 2094909d I fixed it so that `suffix=False` will suppress the suffix (and `suffix=""` still does also).

**Comment by amanagr** [comment]
Thanks @nedbat

**Comment by nedbat** [comment]
This is now released as part of [coverage 6.2](https://pypi.org/project/coverage/6.2).
